data-prepper-plugins/dynamodb-source-coordination-store/src/main/java/org/opensearch/dataprepper/plugins/sourcecoordinator/dynamodb/DynamoDbSourceCoordinationStore.java

```diff
@@ -23,33 +33,106 @@
 @DataPrepperPlugin(name = "dynamodb", pluginType = SourceCoordinationStore.class, pluginConfigurationType = DynamoStoreSettings.class )
 public class DynamoDbSourceCoordinationStore implements SourceCoordinationStore {
 
+    private static final Logger LOG = LoggerFactory.getLogger(DynamoDbSourceCoordinationStore.class);
+
+    static final String AVAILABLE_PARTITIONS_FILTER_EXPRESSION = "contains(sourcePartitionStatus, :s) " +
+            "and (attribute_not_exists(reOpenAt) or reOpenAt = :null or reOpenAt <= :ro) " +
+            "and (attribute_not_exists(partitionOwnershipTimeout) or partitionOwnershipTimeout = :null or partitionOwnershipTimeout <= :t)";
+
     private final DynamoStoreSettings dynamoStoreSettings;
     private final PluginMetrics pluginMetrics;
+    private final DynamoDbClientWrapper dynamoDbClientWrapper;
 
     @DataPrepperPluginConstructor
     public DynamoDbSourceCoordinationStore(final DynamoStoreSettings dynamoStoreSettings, final PluginMetrics pluginMetrics) {
         this.dynamoStoreSettings = dynamoStoreSettings;
         this.pluginMetrics = pluginMetrics;
+        this.dynamoDbClientWrapper = DynamoDbClientWrapper.create(dynamoStoreSettings.getRegion(), dynamoStoreSettings.getStsRoleArn());
+        dynamoDbClientWrapper.tryCreateTable(dynamoStoreSettings.getTableName(), constructProvisionedThroughput(
+                dynamoStoreSettings.getProvisionedReadCapacityUnits(), dynamoStoreSettings.getProvisionedWriteCapacityUnits()));
     }
+
     @Override
     public Optional<SourcePartitionStoreItem> getSourcePartitionItem(final String partitionKey) {
-        return Optional.empty();
+        return dynamoDbClientWrapper.getSourcePartitionItem(partitionKey);
     }
 
     @Override
     public boolean tryCreatePartitionItem(final String partitionKey,
                                           final SourcePartitionStatus sourcePartitionStatus,
                                           final Long closedCount,
                                           final String partitionProgressState) {
+        final DynamoDbSourcePartitionItem newPartitionItem = new DynamoDbSourcePartitionItem();
+        newPartitionItem.setSourcePartitionKey(partitionKey);
+        newPartitionItem.setSourcePartitionStatus(sourcePartitionStatus);
+        newPartitionItem.setClosedCount(closedCount);
+        newPartitionItem.setPartitionProgressState(partitionProgressState);
+        newPartitionItem.setVersion(0L);
+
+        final Optional<SourcePartitionStoreItem> optionalPartitionItem = dynamoDbClientWrapper.getSourcePartitionItem(partitionKey);
+        if (optionalPartitionItem.isEmpty()) {
+             return dynamoDbClientWrapper.tryCreatePartitionItem(newPartitionItem);
+        }
```

Comment:
Yeah true I guess the get is unnecessary due to the conditional check. There should be no concern with race conditions here anyways due to the conditional check, but even if for some reason it did get overwritten, it should be the same item in dynamo since this is for initialization of a partition.